tests/test_at.py

```diff
@@ -0,0 +1,155 @@
+from __future__ import annotations
+
+from contextlib import contextmanager, suppress
+from importlib import import_module
+from typing import TYPE_CHECKING
+
+import numpy as np
+import pytest
+from array_api_compat import (  # type: ignore[import-untyped]  # pyright: ignore[reportMissingTypeStubs]
+    array_namespace,
+    is_dask_array,
+    is_pydata_sparse_array,
+    is_writeable_array,
+)
+
+from array_api_extra import at
+
+if TYPE_CHECKING:
+    from array_api_extra._lib._typing import Array, Untyped
+
+all_libraries = (
+    "array_api_strict",
+    "numpy",
+    "numpy_readonly",
+    "cupy",
+    "torch",
+    "dask.array",
+    "sparse",
+    "jax.numpy",
+)
+
+
+@pytest.fixture(params=all_libraries)
+def array(request: pytest.FixtureRequest) -> Array:
+    library = request.param
+    if library == "numpy_readonly":
+        x = np.asarray([10.0, 20.0, 30.0])
+        x.flags.writeable = False
+    else:
+        try:
+            lib = import_module(library)
+        except ImportError:
+            pytest.skip(f"{library} is not installed")
+        x = lib.asarray([10.0, 20.0, 30.0])
+    return x
+
+
+def assert_array_equal(a: Array, b: Array) -> None:
```

Comment:
`None` or `NoReturn`?

Reply:
None. NoReturn is a function that either always raises or with an unbreakable infinite loop.